test/acceptance/tests.py

```diff
@@ -923,6 +963,45 @@ def test_staff_override_at_beginning(self):
         ])
 
 
+class FullWorkflowRequiredTest(FullWorkflowBaseTest):
+    """
+    Tests of complete workflows, combining multiple required steps together.
+    """
+    def setUp(self):
+        super(FullWorkflowRequiredTest, self).setUp(problem_loc_key="full_workflow_required")
+
+    @retry()
+    @attr('acceptance')
+    def test_train_self_peer_staff(self):
+        """
+        Scenario: complete workflow that included staff required step.
+
+        Given that I have created a submission, completed training, and done a self assessment
+        And a second learner has also created a submission, training, and self assessment
+        Then I can assess a learner
```

Comment:
The other learner is grading this one in the `do_self_train_peer` helper method below.

I've added ddt booleans to this test, they will determine if the other learner actually grades this one or not. This will cover both of the assertions you mentioned:
* peer grading doesn't have to happen to me in order for me to see a score
* a final grade doesn't show until staff grade is done

I've also added some assertions to the `do_staff_assessment` method to make sure the ungraded and checked out numbers are correct, and that "submit and grade another" is working. This should take effect on this test, as we have 2 learners creating a submission.